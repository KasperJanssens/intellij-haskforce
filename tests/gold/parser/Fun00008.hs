module Fun00008 where

f x = let z = if x then 3 else 8
      in 19

main2 = do
            h <- ask
            let m = f h
            13
            return m
