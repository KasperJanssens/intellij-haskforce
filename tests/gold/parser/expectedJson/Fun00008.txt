[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":11,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":11,"srcSpanEndLine":11,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":11,"srcSpanEndLine":11,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":17}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Fun00008"]},null,null]},[],[],[{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[]},[{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":2,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},"f"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":3},"srcInfoPoints":[]},"x"]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":5}]},{"Let":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":4,"srcSpanStartColumn":7},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":7},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":7}]},{"BDecls":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":3,"srcSpanEndLine":4,"srcSpanStartColumn":11},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":7}]},[{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},"z"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13}]},{"If":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":15},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":15},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},"x"]}]}]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":25},"srcInfoPoints":[]},3,"3"]}]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":32},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":32},"srcInfoPoints":[]},8,"8"]}]}]}]},null]}]]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},19,"19"]}]}]}]},null]}]]},{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":6,"srcSpanEndLine":11,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[]},"main2"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":6,"srcSpanEndLine":11,"srcSpanStartColumn":7},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":7}]},{"Do":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":6,"srcSpanEndLine":11,"srcSpanStartColumn":9},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":9},{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":13},{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":13},{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":11,"srcSpanEndLine":11,"srcSpanStartColumn":1}]},[{"Generator":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":13},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":15}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":13},"srcInfoPoints":[]},"h"]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},"ask"]}]}]}]},{"LetStmt":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":8,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":13}]},{"BDecls":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":8,"srcSpanEndLine":9,"srcSpanStartColumn":17},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":17},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13}]},[{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":17},"srcInfoPoints":[]},"m"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":19},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":19}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":21},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":21},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":21},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":21},"srcInfoPoints":[]},"f"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":23},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":23},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":23},"srcInfoPoints":[]},"h"]}]}]}]}]},null]}]]}]},{"Qualifier":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},13,"13"]}]}]},{"Qualifier":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":13},"srcInfoPoints":[]},"return"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":20},"srcInfoPoints":[]},"m"]}]}]}]}]}]]}]},null]}]]},[]]