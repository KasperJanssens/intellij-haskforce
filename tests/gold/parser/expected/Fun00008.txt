Haskell File
  HaskellPpragmaImpl(PPRAGMA)
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)(' LANGUAGE TupleSections ')
    PsiElement(HaskellTokenType.#-})('#-}')
  HaskellModuledeclImpl(MODULEDECL)
    PsiElement(HaskellTokenType.module)('module')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Fun00008')
    PsiElement(HaskellTokenType.where)('where')
  HaskellBodyImpl(BODY)
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    PsiElement(HaskellTokenType.let)('let')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('z')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    PsiElement(HaskellTokenType.if)('if')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    PsiElement(HaskellTokenType.then)('then')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    PsiElement(HaskellTokenType.integertoken)('3')
    HaskellSymbolImpl(SYMBOL)
      PsiElement(HaskellTokenType.^)('^')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    PsiElement(HaskellTokenType.else)('else')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellSymbolImpl(SYMBOL)
      PsiElement(HaskellTokenType.^)('^')
    PsiElement(HaskellTokenType.integertoken)('8')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    PsiElement(HaskellTokenType.in)('in')
    HaskellSymbolImpl(SYMBOL)
      PsiElement(HaskellTokenType.-)('-')
    PsiElement(HaskellTokenType.integertoken)('1')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('main2')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    PsiElement(HaskellTokenType.do)('do')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('h')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.<-)('<-')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('ask')
    PsiElement(HaskellTokenType.let)('let')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('m')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('h')
    PsiElement(HaskellTokenType.integertoken)('13')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    PsiElement(HaskellTokenType.integertoken)('3')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('return')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('m')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('g')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        PsiElement(HaskellTokenType.Varsym)('++')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('ys')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        PsiElement(HaskellTokenType.Varsym)('++')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('zs')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        PsiElement(HaskellTokenType.Varsym)('++')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('zs'')
    PsiElement(HaskellTokenType.where)('where')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    PsiElement(HaskellTokenType.integertoken)('1')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType...)('..')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('ys')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    PsiElement(HaskellTokenType.integertoken)('1')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType...)('..')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('zs')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    PsiElement(HaskellTokenType.integertoken)('1')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType...)('..')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('zs'')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    PsiElement(HaskellTokenType.integertoken)('1')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType...)('..')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    PsiElement(HaskellTokenType.integertoken)('30')
    HaskellSymbolImpl(SYMBOL)
      PsiElement(HaskellTokenType.*)('*')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('h')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Just')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.<-)('<-')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        PsiElement(HaskellTokenType.Varsym)('++')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('z')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Just')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.<-)('<-')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('z')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.<-)('<-')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')